rest/server_context.go

```diff
@@ -946,6 +954,92 @@ func (sc *ServerContext) getDbConfigFromServer(dbName string) (*DbConfig, error)
 	return &config, nil
 }
 
+//////// STATS LOGGING
+
+func (sc *ServerContext) startStatsLogger() {
+
+	statsLogFrequencySecs := 5
+	if sc.config.Unsupported != nil && sc.config.Unsupported.Logging != nil {
+		statsLogFrequencySecs = sc.config.Unsupported.Logging.StatsLogFrequencySecs
+	}
+
+	interval := time.Second * time.Duration(statsLogFrequencySecs)
+
+	sc.statsLoggingTicker = time.NewTicker(interval)
+	go func() {
+		for range sc.statsLoggingTicker.C {
+			sc.logStats()
+		}
+	}()
+	base.Infof(base.KeyAll, "Logging stats with frequency: %v", interval)
+
+}
+
+func (sc *ServerContext) stopStatsLogger() {
+	if sc.statsLoggingTicker != nil {
+		sc.statsLoggingTicker.Stop()
+	}
+}
+
+func (sc *ServerContext) logStats() {
+
+	AddGoRuntimeStats()
+
+	// Create wrapper expvar map in order to add a timestamp field for logging purposes
+	logStats := new(expvar.Map).Init()
+	logStats.Set("stats", base.Stats)
+	logStats.Set("unix_epoch_timestamp", base.ExpvarInt64Val(time.Now().Unix()))
```

Comment:
What's the advantage in using a wrapper expvar here, as opposed to having a simple struct that gets marshalled as json?

Reply:
I'd attempted this, and it "didn't work".  Just retried it to refresh my memory, and this code:

```
func (sc *ServerContext) logStats() error {

	AddGoRuntimeStats()

	wrapper := struct{
		Stats *expvar.Map `json:"stats"`
		Unix_epoch_timestamp int64 `json:"unix_epoch_timestamp"`
	}{
		Stats: base.Stats,
		Unix_epoch_timestamp: time.Now().Unix(),
	}

	base.RecordStats([]byte(base.Stats.String()))

	marshalled, err := json.Marshal(wrapper)
	if err != nil {
		return err
	}

	base.RecordStats(marshalled)

	return nil


}
```

outputs:

```
{"global": {"resource_utilization": {"go_memstats_heapalloc": 4002744, "go_memstats_heapidle": 32768, "go_memstats_heapinuse": 5242880, "go_memstats_heapreleased": 0, "go_memstats_pausetotalns": 411581, "go_memstats_stackinuse": 884736, "go_memstats_stacksys": 884736, "go_memstats_sys": 10033400, "memory_rss_bytes": 10033400, "num_goroutines": 12}}, "per_db": {}, "per_replication": {}}
{"stats":{},"unix_epoch_timestamp":1543438439}
```

Note the empty `"stats":{},"`.  I believe it's because expvar.Map doesn't implement the `json.Marhsal()` interface.  Let me know if you see something I'm doing wrong.

Reply:
I was thinking the stats would be included in the struct as `json.RawMessage`, and populated via  struct.stats = expvar.Map.String().  This still feels like less of a hack to me than using a transient expvar.Map that doesn't get published.

Reply:
This worked.  Updated.